### PR TITLE
Disallow aggregation function as `UNNEST` parameter

### PR DESCRIPTION
**Pull request by radek-starburst**
## Description
Disallow aggregation function as UNNEST parameter and add similar check as here: https://github.com/trinodb/trino/pull/17988/commits/ad637f702ef272f4b96eca006fb047379f4d0604#diff-bd8fefc1e4638b7f65e377b75c16b46a2c541a24a64993860663507351917230R106

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/metadata/InternalFunctionBundle.java [hunk 1]
```diff
@@ -138,8 +138,9 @@ public ScalarFunctionImplementation getScalarFunctionImplementation(
 
     private SpecializedSqlScalarFunction specializeScalarFunction(FunctionId functionId, BoundSignature boundSignature, FunctionDependencies functionDependencies)
     {
-        SqlScalarFunction function = (SqlScalarFunction) getSqlFunction(functionId);
-        return function.specialize(boundSignature, functionDependencies);
+        SqlFunction function = getSqlFunction(functionId);
+        checkArgument(function instanceof SqlScalarFunction, "%s is not a scalar function", function.getFunctionMetadata().getSignature());
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/metadata/InternalFunctionBundle.java:142]: We need unit tests for that and this is an SQL error so we need a proper UX error.

Comment (radek-starburst) [reply core/trino-main/src/main/java/io/trino/metadata/InternalFunctionBundle.java:142]: On my eye it is not SQL error. It is kind of invariant that should be satisfied here. We have similar here: 
https://github.com/trinodb/trino/commit/ad637f702ef272f4b96eca006fb047379f4d0604#diff-bd8fefc1e4638b7f65e377b75c16b46a2c541a24a64993860663507351917230R106
Are sure that we should test every invariant? This invariant is to help detect such problem immediately in the future. But, the end user should observe this (if there is no bug in code).

Comment (electrum) [reply core/trino-main/src/main/java/io/trino/metadata/InternalFunctionBundle.java:142]: This should already be handled during SQL analysis by `io.trino.sql.analyzer.Analyzer#verifyNoAggregateWindowOrGroupingFunctions` as the newly added test shows.

File: core/trino-main/src/main/java/io/trino/metadata/InternalFunctionBundle.java
Comment (kasiafi) [inline core/trino-main/src/main/java/io/trino/metadata/InternalFunctionBundle.java:1]: In my opinion, we can leave those to fail with `ClassCastException`. It is equally easy to inspect.

File: core/trino-main/src/main/java/io/trino/metadata/InternalFunctionBundle.java (hunk 1 above)
Comment (radek-starburst) [reply core/trino-main/src/main/java/io/trino/metadata/InternalFunctionBundle.java:142]: Yes, this case is handled by `StatementAnalyzer` now. This is why I've put assertions, as the last ressort. [The first PR did not include `verifyNoAggregateWindowOrGroupingFunctions` so it was true then that it was exposed to the end user].

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java [hunk 2]
```diff
@@ -6681,6 +6681,14 @@ public void testJsonTable()
                 .hasMessage("line 1:15: JSON_TABLE is not yet supported");
     }
 
+    @Test
+    public void testDisallowAggregationFunctionInUnnest()
+    {
+        assertFails("SELECT a from (values (1), (2)) t(a), unnest(ARRAY[count(t.a)])")
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java:6687]: use capital letters for keywords like `FROM`

**Review by kokosing — APPROVED** [review]

**Review by kokosing — APPROVED** [review]

**Pull request merged**